core/trino-main/src/main/java/io/trino/execution/SessionEvaluator.java

```diff
@@ -0,0 +1,102 @@
+import io.trino.sql.PlannerContext;
+import io.trino.sql.tree.Expression;
+import io.trino.sql.tree.NodeRef;
+import io.trino.sql.tree.Parameter;
+import io.trino.sql.tree.QualifiedName;
+
+import java.util.List;
+import java.util.Map;
+
+import static io.trino.metadata.MetadataUtil.getRequiredCatalogHandle;
+import static io.trino.metadata.SessionPropertyManager.evaluatePropertyValue;
+import static io.trino.metadata.SessionPropertyManager.serializeSessionProperty;
+import static io.trino.spi.StandardErrorCode.INVALID_SESSION_PROPERTY;
+import static io.trino.sql.analyzer.SemanticExceptions.semanticException;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class SessionEvaluator
+{
+    private final PlannerContext plannerContext;
+    private final AccessControl accessControl;
+    private final SessionPropertyManager sessionPropertyManager;
+
+    @Inject
+    public SessionEvaluator(PlannerContext plannerContext, AccessControl accessControl, SessionPropertyManager sessionPropertyManager)
+    {
+        this.plannerContext = requireNonNull(plannerContext, "plannerContext is null");
+        this.accessControl = requireNonNull(accessControl, "accessControl is null");
+        this.sessionPropertyManager = requireNonNull(sessionPropertyManager, "sessionPropertyManager is null");
+    }
+
+    public String evaluate(Session session, QualifiedName name, Expression expression, Map<NodeRef<Parameter>, Expression> parameters)
+    {
+        List<String> nameParts = name.getParts();
+        if (nameParts.size() == 1) {
+            PropertyMetadata<?> systemPropertyMetadata = sessionPropertyManager.getSystemSessionPropertyMetadata(nameParts.getFirst())
+                    .orElseThrow(() -> semanticException(INVALID_SESSION_PROPERTY, expression, "Session property %s does not exist", name));
+
+            return toSessionValue(session, name, expression, parameters, systemPropertyMetadata);
+        }
+        else if (nameParts.size() == 2) {
+            String catalogName = nameParts.getFirst();
+            String propertyName = nameParts.getLast();
+
+            CatalogHandle catalogHandle = getRequiredCatalogHandle(plannerContext.getMetadata(), session, expression, catalogName);
+            PropertyMetadata<?> connectorPropertyMetadata = sessionPropertyManager.getConnectorSessionPropertyMetadata(catalogHandle, propertyName)
+                    .orElseThrow(() -> semanticException(INVALID_SESSION_PROPERTY, expression, "Session property %s does not exist", name));
+
+            return toSessionValue(session, name, expression, parameters, connectorPropertyMetadata);
+        }
+        throw semanticException(INVALID_SESSION_PROPERTY, expression, "Invalid session property '%s'", name);
+    }
+
+    private String toSessionValue(Session session, QualifiedName name, Expression expression, Map<NodeRef<Parameter>, Expression> parameters, PropertyMetadata<?> propertyMetadata)
```

Comment:
`toSessionValue` doesn't mean convey the right concept. This method is actually no different than the one above, except that it includes the metadata of the property. Rename it to `evaluate`.